src/main/java/io/lettuce/core/RedisAuthenticationHandler.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019-Present, Redis Ltd. and Contributors
+ * All rights reserved.
+ *
+ * Licensed under the MIT License.
+ *
+ * This file contains contributions from third-party contributors
+ * licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lettuce.core;
+
+import io.lettuce.core.cluster.pubsub.StatefulRedisClusterPubSubConnection;
+import io.lettuce.core.protocol.ProtocolVersion;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * Handles reauthentication of a connection each time a new authentication token is provided by
+ * `RenewableRedisCredentialsProvider`.
+ * <p>
+ * This class is part of the internal API.
+ *
+ * @author Ivo Gaydajiev
+ */
+class RedisAuthenticationHandler extends BaseRedisAuthenticationHandler<StatefulRedisConnectionImpl<?, ?>> {
```

Comment:
looks like we have this `BaseRedisAuthenticationHandler` class missing ?

Reply:
Yes it is, I am planning (wotking) on separating the PR in two, 
1) Changes needed to support StreamingCredentialProvider hence (reauthenticate )
   this can be used with rotating credentials, or token based auth with expiring credentials
  
2) Dedicated PR adding  TokenBasedCredentialsManager using the core library + example how to use it with concreate auth provider

Reply:
Rebased the PR. Should be available now.